src/economy/currency.ts

```diff
@@ -8,74 +9,113 @@ export const Currency = {
 } as const;
 export type Currency = typeof Currency[keyof typeof Currency];
 
-interface CurrencyName {
+interface CurrencyTranslation {
   unit: string;
   unitPlural: string;
   subunit: string;
   subunitPlural: string;
 }
 
-interface CurrencyNames {
-  translations: CurrencyName;
-  plainText: CurrencyName;
-}
-
-const currencyNames = new Map<Currency, CurrencyNames>([
+const CURRENCY_TRANSLATIONS = new Map<Currency, CurrencyTranslation>([
   [
     Currency.CLASSIC_GOLD,
     {
-      translations: {
-        unit: 'vk.currency_a_unit',
-        unitPlural: 'vk.currency_a_unit_plural',
-        subunit: 'vk.currency_a_subunit',
-        subunitPlural: 'vk.currency_a_subunit_plural',
-      },
-      plainText: {
-        unit: 'Punta',
-        unitPlural: 'Puntaa',
-        subunit: 'Penni',
-        subunitPlural: 'Penniä',
-      },
+      unit: 'vk.currency_a_unit',
+      unitPlural: 'vk.currency_a_unit_plural',
+      subunit: 'vk.currency_a_subunit',
+      subunitPlural: 'vk.currency_a_subunit_plural',
     },
   ],
   [
     Currency.SILVER,
     {
-      translations: {
-        unit: 'vk.currency_b_unit',
-        unitPlural: 'vk.currency_b_unit_plural',
-        subunit: 'vk.currency_b_subunit',
-        subunitPlural: 'vk.currency_b_subunit_plural',
-      },
-      plainText: {
-        unit: 'Sillinki',
-        unitPlural: 'Sillinkiä',
-        subunit: 'Sentti',
-        subunitPlural: 'Senttiä',
-      },
+      unit: 'vk.currency_b_unit',
+      unitPlural: 'vk.currency_b_unit_plural',
+      subunit: 'vk.currency_b_subunit',
+      subunitPlural: 'vk.currency_b_subunit_plural',
     },
   ],
   [
     Currency.GREEN_PAPER,
     {
-      translations: {
-        unit: 'vk.currency_c_unit',
-        unitPlural: 'vk.currency_c_unit_plural',
-        subunit: 'vk.currency_c_subunit',
-        subunitPlural: 'vk.currency_c_subunit_plural',
-      },
-      plainText: {
-        unit: 'Rupla',
-        unitPlural: 'Ruplaa',
-        subunit: 'Kopeekka',
-        subunitPlural: 'Kopeekkaa',
-      },
+      unit: 'vk.currency_c_unit',
+      unitPlural: 'vk.currency_c_unit_plural',
+      subunit: 'vk.currency_c_subunit',
+      subunitPlural: 'vk.currency_c_subunit_plural',
     },
   ],
 ]);
 
-export function getCurrencyNames(currency: Currency) {
-  return currencyNames.get(currency);
+const FALLBACK_TRANSLATION: CurrencyTranslation = {
+  unit: 'vk.currency_unknown_unit',
+  unitPlural: 'vk.currency_unknown_unit_plural',
+  subunit: 'vk.currency_unknown_subunit',
+  subunitPlural: 'vk.currency_unknown_subunit_plural',
+};
+
+/**
```

Comment:
If these translations are in resource pack anyway, why do we need them here? Sign support for translated components is known to be possible and planned (made an issue Valtakausi/craftjs#47).

Reply:
These are easier to handle in the code since they are strings and constructing new sentences becomes cleaner because we don't have to insert TranslatableComponents in the middle.
And there is a small difference: In resource pack "%s Pounds" and in TS "Pounds". Resource pack translations are useful for item names and for signs, but translated strings are more useful when formatting the chat messages.